### ISSUE TITLE
关于TimesNet中FFT_for_Period函数可能有误的问题

### ISSUE THREAD

**Issue by Tang4109** [thuml/Time-Series-Library#157]
def FFT_for_Period(x, k=2):
    # [B, T, C]
    xf = torch.fft.rfft(x, dim=1)
    # find period by amplitudes
    frequency_list = abs(xf).mean(0).mean(-1)
    frequency_list[0] = 0
    _, top_list = torch.topk(frequency_list, k)
    top_list = top_list.detach().cpu().numpy()
    period = x.shape[1] // top_list
    return period, abs(xf).mean(-1)[:, top_list]

倒数第二行在计算周期的时候
period = x.shape[1] // top_list
我理解的是信号长度除以振幅最高的k个频率得到对应周期长度，但是这里top_list是索引，是否应该改为索引对应的频率？
不知道我理解的对不对

**Comment by rdyan0053** [comment]
我感觉代码有错啊，为什么我用心电数据，计算出来的top_list里面的值是`[1,2,3]`(我将超参数k设置为3)。这也太离谱了，难道我的数据通过fft转换之后，振幅最大的三个对应的频率是1,2,3?

另外，如果top_list是频率，周期不是通过频率倒数得到的吗？
